projects/common/src/telemetry/track/track.directive.ts

```diff
@@ -0,0 +1,60 @@
+import { fromEvent, Subscription } from 'rxjs';
+import { Directive, ElementRef, Input, OnChanges, OnDestroy } from '@angular/core';
+import { TypedSimpleChanges } from '../../utilities/types/angular-change-object';
+import { TrackUserEventsType } from '../telemetry';
+import { UserTelemetryImplService } from '../user-telemetry-impl.service';
+
+@Directive({
+  selector: '[htTrack]'
+})
+export class TrackDirective implements OnChanges, OnDestroy {
+  @Input('htTrack')
+  public userEvents: TrackUserEventsType[] = [TrackUserEventsType.Click];
+
+  @Input('htTrackLabel')
+  public label?: string;
+
+  private readonly subscription: Subscription = new Subscription();
+  private trackedEventLabel: string = '';
+
+  public constructor(
+    private readonly host: ElementRef,
+    private readonly userTelemetryImplService: UserTelemetryImplService
+  ) {}
+
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.userEvents) {
+      this.setupListeners();
+    }
+
+    if (changes.label) {
+      this.trackedEventLabel = this.label ?? (this.host.nativeElement as HTMLElement)?.tagName;
```

Comment:
tag name doesnt seem very useful, maybe we should require label? Could make that the main arg and the event name the addtl arg since that's easier to default.

Reply:
So, the autotrack based telemetry tools extract certain fields from the element. I am yet to figure out the relevant ones. After I do that, I will add those fields here. 

Imo htTrack= "events" is more readable.  You can always miss providing a directive input and angular won't complain. So let's keep the default logic. I will use a better field than just <tagname> with a later PR.